src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -136,7 +149,11 @@ extends AbstractMap[Key, Value]
       None
     } else {
       val res = entry.value
```

Comment:
> in `remove`, the `key` should be nulled out, to avoid memory leaks

OK. That's not how I understood your earlier comment. I haven't seen this discussed anywhere else yet. It also didn't occur to me, but I've been skeptical of that whole business of keeping the old `Entry`s around.

If it's contentious, people can hash it out :grin: on [SI-9522](https://issues.scala-lang.org/browse/SI-9522).